File: core/src/dto/task.rs
mod list_tasks;
mod update_task;

use chrono::{DateTime, Utc};
use uuid::Uuid;

use crate::model::{Task, TaskStatus};

pub use list_tasks::ListTasksRequest;
pub use update_task::UpdateTaskRequest;

#[derive(Debug, Clone)]
pub struct TaskResponse {
    pub id: Uuid,
    pub repo_owner: String,
    pub repo_name: String,
    pub build_id: Uuid,
    pub name: String,
    pub command: String,
    pub status: TaskStatus,
    pub waits_for: Vec<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl From<Task> for TaskResponse {
    fn from(task: Task) -> Self {
        Self {
            id: task.id,
            repo_owner: task.repo_owner,
            repo_name: task.repo_name,
            build_id: task.build_id,
            name: task.name,
            command: task.command,
            status: task.status,
            waits_for: task.waits_for,
            created_at: task.created_at,
            updated_at: task.updated_at,
        }
    }
}

File: frontend/app/lib/dto/task.ts
import { z } from "zod";

export const CreateTaskRequestSchema = z.object({
  repo_owner: z.string(),
  repo_name: z.string(),
  command: z.string(),
});

export type CreateTaskRequest = z.infer<typeof CreateTaskRequestSchema>;

export const TaskResponseSchema = z.object({
  id: z.uuid(),
  repo_owner: z.string(),
  repo_name: z.string(),
  build_id: z.uuid().optional(),
  name: z.string().optional(),
  command: z.string(),
  status: z.string(),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
});

export type TaskResponse = z.infer<typeof TaskResponseSchema>;

export const TasksResponseSchema = z.array(TaskResponseSchema);

export type TasksResponse = z.infer<typeof TasksResponseSchema>;
